src/js/angular/repositories/controllers.js

```diff
@@ -469,12 +485,24 @@ function AddRepositoryCtrl($scope, toastr, $repositories, $location, Upload, isE
         });
     };
 
-    if ($scope.isEnterprise) {
-        $scope.getConfig('worker');
-    } else if ($scope.isFreeEdition) {
-        $scope.getConfig('free');
+    $scope.isRepositoryTypeValid = function () {
+        if ($scope.isEnterprise && ($scope.repositoryType === REPOSITOTY_TYPES.eeMaster
```

Comment:
I'd have created some functions with meaningful names for these checks, e.g. `isValidEERepository()`, `isSERepository()`, `isValidFERepository()`

Reply:
I extracted logic in separate methods.